src/_json.js

```diff
@@ -16,6 +16,36 @@ function parseJSON(json) {
   return JSON.parse(json, json_parse)
 }
 
+function parseJSONStreaming(json) {
+  var results = []
+
+  try {
+    results.push(parseJSON(json))
+  } catch (error) {
+    var position
+
+    /** JSON parse error message is throw by the browser */
+    if (error.message.match(/JSON.parse/g)) {
+      var matchResult = error.message.match(/column ([0-9+]*)/)
+      position = matchResult[1] - 1
```

Comment:
We should avoid multiple regex operations where possible; this code could get called very frequently.

```suggestion
    var capture
    if (capture = error.message.match(/JSON.parse: unexpected non-whitespace.*column (\d+)/)) {
      position = capture[1] - 1
```

Reply:
In a browser at least, `.indexOf()` is 10 times faster than a regex, so I would suggest this instead.  We test FOR the error quickly, then use the more expensive operation to get the position if it's happened

if (error.message.indexOf('JSON.parse')) {
      var matchResult = error.message.match(/column ([0-9+]*)/)

Reply:
https://www.measurethat.net/Benchmarks/Show/975/11/regex-vs-indexof-vs-startswith-vs-substr#latest_results_block

Reply:
This regex is incorrect as well

use `/column ([0-9]+)/`

Reply:
> This regex is incorrect as well
> 
> use `/column ([0-9]+)/`

It is not. `column ([0-9]+)` didn't work. 😕

Reply:
Ok, I'm going to use `indexOf` to check the error.

Reply:
@danieltodonnell How is the regex that I suggested incorrect? I tested it against the error message quoted in the issue description. I might be overlooking regex support in browsers that the driver supports, so I'm curious what I'm missing.

Reply:
Are we sure the message is stable across browsers + node js versions?

Reply:
@erickpintor I was only able to catch these 2 messages but I'm not 100% sure these are the only possible messages.